### PR TITLE
project roles: make adding groups work

### PR DESCRIPTION
**Pull request by valentijnscholten**
The Role (project role) resource has a method add_user that, according to its documentation and parameters, should be able to add either users or groups to a project role.
The adding of groups was not (yet) implemented. This PR implements that.

Please note that JIRA server only supports either adding users XOR groups in a single request. So I have added a quick check to raise a clear error. The error message from JIRA is some vague 400 error about it being unable to parse the json request.

it's not perfect, but at least the method now works according to its docs.

### CODE REVIEW

**Comment by studioj** [comment]
@valentijnscholten please merge in master, solve conflicts

**Comment by studioj** [comment]
thx => CI is running, if green it'll get reviewed

codacy is failing atm because of this
https://app.codacy.com/gh/pycontribs/jira/pullRequest?prid=7435647
black is failing on resources.py
https://github.com/pycontribs/jira/pull/884/checks?check_run_id=2657924451#step:8:47

File: jira/resources.py [hunk 1]
```diff
@@ -956,13 +956,19 @@ def add_user(
             groups (Optional[Union[str,List,Tuple]]): a group or groups to add to the role
         """
 
+        data = None
         if users is not None and isinstance(users, str):
             users = (users,)
+            data = {"user": users}
+            if groups is not None:
+                raise NotImplementedError("JIRA can only add users XOR groups")
```
Comment (adehad) [inline jira/resources.py:964]:
```suggestion
        if users is not None: 
            if isinstance(users, str):
                users = (users,)
            data = {"user": users}
            if groups is not None:
                raise NotImplementedError("JIRA can only add users XOR groups")
```

This function accepts both strings and a list/tuple already, so we need to be a bit smarter on the handling here

File: jira/resources.py [hunk 2]
```diff
@@ -956,13 +956,19 @@ def add_user(
...
+
         if groups is not None and isinstance(groups, str):
             groups = (groups,)
+            data = {"group": groups}
```
Comment (adehad) [inline jira/resources.py:968]:
```suggestion
        if groups is not None:
            if isinstance(groups, str):
                groups = (groups,)
            data = {"group": groups}
```

as above

**Review by adehad — CHANGES_REQUESTED** [review]

**Comment by studioj** [comment]
stale for 4 months => close for now

**Pull request closed without merging**